heksher/heksher_client.py

```diff
@@ -0,0 +1,71 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from typing import Mapping, Collection
+from weakref import WeakSet
+
+import heksher.main_client
+from heksher.setting import Setting
+
+
+class BaseHeksherClient(ABC):
+    @abstractmethod
+    def handover_main(self, other: BaseHeksherClient):
+        """
+        attempt to hand over the "main"-ness of the client from self to other. Might fail if self feels this
+        shouldn't happen. Calling this when self is not the main client will result in a runtime error. Note that this
+        method should only handle the stopping and cleanup of self, and not start other.
+        """
+        if heksher.main_client.Main is not self:
```

Comment:
I think Handover should support only getting a temproary client, as it's not like you can transfer between heksher clients